runtime/stdlib/contract_update_validation.go

```diff
@@ -215,6 +217,30 @@ func (validator *ContractUpdateValidator) hasErrors() bool {
 	return len(validator.errors) > 0
 }
 
+func collectRemovedTypePragmas(validator UpdateValidator, pragmas []*ast.PragmaDeclaration) (removedTypes []ast.Identifier) {
```

Comment:
Maybe make the result set. It is only used for inclusion tests (contains), right? Maybe the AST identifier's identifier (string) is enough?

Also consider that `ast.Identifier` may have nested identifiers. Those should be maybe rejected in `collectRemovedTypePragmas`. e.g. `#removedType(Foo.Bar)`

Reply:
> Also consider that ast.Identifier may have nested identifiers. Those should be maybe rejected in collectRemovedTypePragmas. e.g. #removedType(Foo.Bar)

Yeah, it's already rejected. `ast.NominalType` can have nested identifiers, but `ast.Identifer` is just a wrapper around a string, i.e. no nested identifiers. We check in `collectRemovedTypePragmas` that the argument to `#removedType` is an identifier, restricting it to just `Foo`-shaped things

Reply:
Ah right, I confused `ast.Identifer` with `ast.NominalType` 👍